validator/src/main.rs

```diff
@@ -2666,91 +2768,88 @@ pub fn main() {
-                .expect("failed to set tls_config");
-        }
-
-        let addr: String = value_of(&matches, "block_engine_url").unwrap_or_else(|| {
-            value_of(&matches, "block_engine_address").expect("missing block-engine-url")
-        });
-        let mut backend_endpoint = Endpoint::from_shared(addr.clone())
-            .expect("invalid block-engine-address value")
-            .tcp_keepalive(Some(Duration::from_secs(60)));
-        if addr.contains("https") {
-            backend_endpoint = backend_endpoint
-                .tls_config(tonic::transport::ClientTlsConfig::new())
-                .expect("failed to set tls_config");
-        }
-
-        BlockEngineConfig {
-            auth_service_endpoint,
-            backend_endpoint,
-            trust_packets: matches.is_present("trust_block_engine_packets"),
-        }
-    });
-
-    let is_relayer_enabled = matches.is_present("relayer_url")
-        || matches.is_present("relayer_auth_service_address")
-        || matches.is_present("relayer_address")
-        || matches.is_present("trust_relayer_packets")
-        || matches.is_present("relayer_expected_heartbeat_interval_ms")
-        || matches.is_present("relayer_max_failed_heartbeats");
-    let maybe_relayer_config = is_relayer_enabled.then(|| {
-        let addr: String = value_of(&matches, "relayer_url").unwrap_or_else(|| {
-            value_of(&matches, "relayer_auth_service_address").expect("missing relayer-url")
-        });
-        let mut auth_service_endpoint =
-            Endpoint::from_shared(addr.clone()).expect("invalid relayer-url value");
-        if addr.contains("https") {
-            auth_service_endpoint = auth_service_endpoint
-                .tls_config(tonic::transport::ClientTlsConfig::new())
-                .expect("failed to set tls_config");
+    let mut block_engine_config = BlockEngineConfig::default();
+    if matches.is_present("block_engine_url") {
+        block_engine_config.block_engine_url =
+            value_of(&matches, "block_engine_url").expect("couldn't parse block_engine_url");
+    } else {
+        let error_msg = "Specifying seperate auth and backend addresses for block engine is deprecated.  Recommended to use --block_engine_url instead.\
+                    If using block_engine_auth_service_address and block_engine_address, they must both be provided and set to the same value.";
+        match (
+            matches.is_present("block_engine_auth_service_address"),
+            matches.is_present("block_engine_address"),
+        ) {
+            (true, true) => {
+                let auth_addr: String = value_of(&matches, "block_engine_auth_service_address")
+                    .expect("couldn't parse block_engine_auth_service_address");
+                let backend_addr: String = value_of(&matches, "block_engine_address")
+                    .expect("couldn't parse block_engine_address");
+                if auth_addr != backend_addr {
```

Comment:
this might be breaking behavior as some validators might still be providing the auth + vifc addresses separately with ports attached. 

this is probably fine for master, but ideally in v1.14 we keep the behavior the same